components/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/internal/TenantServiceCreator.java

```diff
@@ -17,7 +17,7 @@
 *
 */
 
-package org.wso2.carbon.appmgt.impl.observers;
+package org.wso2.carbon.appmgt.gateway.internal;
```

Comment:
Remove the registry related functions from the gateway and put them to the Impl from a separate obsever class